src/qiskit_qec/circuits/code_circuit.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Class that manage circuits for codes."""
+from abc import ABC, abstractmethod
+
+
+class CodeCircuit(ABC):
+    """
+    Abstract class to manage circuits for codes, as well
+    as other fault-tolerant circuits.
+
+    A CodeCircuit requires the methods `string2nodes`,
+    `check_nodes` and `is_cluster_neutral` in order to
+    interface with its `DecodingGraph` and decoders.
+    """
+
+    def __init__(self):
```

Comment:
I would add some more general properties that every circuit has, such as d, T, the basis and have the abstract class set these.

Reply:
Good idea. I added d and T to the description. I think basis is not so universal (some have more complex relationships with bases).